contracts/utils/TimestampedHashRegistry.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/ITimestampedHashRegistry.sol";
+
+// TODO: would SelfMulticall be useful?
+contract TimestampedHashRegistry is Ownable, EIP712, ITimestampedHashRegistry {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _hashTypeToSigners;
+
+    mapping(bytes32 => SignedHash) public hashTypeToSignedHash;
+
+    bytes32 private constant _SIGNED_HASH_TYPE_HASH =
+        keccak256(
+            "SignedHash(bytes32 hashType,bytes32 hash,uint256 timestamp)"
+        );
+
+    constructor() EIP712("TimestampedHashRegistry", "1.0.0") {}
+
+    function setHashTypeSigners(
+        bytes32 hashType,
+        address[] calldata signers
+    ) external onlyOwner {
+        require(hashType != bytes32(0), "Hash is zero");
+        require(signers.length != 0, "Signers length is empty");
+        for (uint256 ind = 0; ind < signers.length; ind++) {
+            _hashTypeToSigners[hashType].add(signers[ind]);
```

Comment:
Say I call `setHashTypeSigners(typeA, [signerA, signerB, signerC])` and then I call `setHashTypeSigners(typeA, [signerC, signerD, signerE])`. The resulting signer set will be `[signerA, signer B, signerC, signerD, signerE]`, which I suspect is not what you intend.